smoker/server/daemon.py

```diff
@@ -164,9 +162,33 @@ def run(self):
             signal.signal(signal.SIGTERM, self._shutdown)
         # API server loop now runs in a separate process and we don't want
         # to terminate to keep an instance of pluginmanager
-        while True:
-            time.sleep(1)
 
+        self._watchdog()
+
+    def _watchdog(self):
+        lg.debug("Starting the smoker watchdog")
+        setproctitle.setproctitle("smokerd watchdog")
```

Comment:
Yes, you're right. 
Starting the watchdog as a separate process is not an option since only the parent process is capable of collecting return codes of dead child processes. But I definitely can keep it the original name.